### PR TITLE
Added orientation to GPS using Compass

### PR DESCRIPTION
**Pull request by gabryelreyes**

### CODE REVIEW

File: lib/HALSim/src/Gps.cpp [hunk 1]
```diff
@@ -94,6 +97,36 @@ bool Gps::getPosition(int32_t& xPos, int32_t& yPos)
...
+
+    if (nullptr != m_compass)
+    {
+        /* Get orientation in radians from Webots. */
+        const double* orientationVector = m_compass->getValues();
+
+        if (nullptr != orientationVector)
+        {
+            double xVector = orientationVector[0];
+            double yVector = orientationVector[1];
```
Comment (jkerpe) [inline lib/HALSim/src/Gps.cpp:112]: The names `xVector` and `yVector` are confusing --> sound multi-dimensional

Comment (gabryelreyes) [reply lib/HALSim/src/Gps.cpp:112]: It is multi-dimensional. X and Y directions :D

Comment (jkerpe) [reply lib/HALSim/src/Gps.cpp:112]: `xVector` and `yVector` are one-dimensional variables, but the `vector` inside the name make them sound multi-dimensional --> maybe something with `component` or `direction`?

File: lib/HALSim/src/Gps.cpp [hunk 2]
```diff
@@ -55,17 +58,17 @@
  * Local Variables
  *****************************************************************************/
 
-/** Index of the x-coordinate in the position vector. */
-static const size_t GPS_POSITION_X_INDEX = 0U;
+/** Index of the x-component in the a vector. */
```
Comment (jkerpe) [inline lib/HALSim/src/Gps.cpp:61]: "in **the a** vector"

File: lib/HALSim/src/Gps.cpp [hunk 3]
```diff
@@ -55,17 +58,17 @@
...
+static const size_t X_COMPONENT_INDEX = 0U;
 
-/** Index of the y-coordinate in the position vector. */
-static const size_t GPS_POSITION_Y_INDEX = 1U;
+/** Index of the y-component in the a vector. */
```
Comment (jkerpe) [inline lib/HALSim/src/Gps.cpp:64]: "in **the a** vector"

**Review by jkerpe — APPROVED** [review]

**Pull request merged**